Add support for Podman

ccpcfrdc opened the pull request:
Add support for Podman containers as well.

apollo13 commented:
With podman & systemd it is also common to run with `--cgroups=split` in which case you'd have:

`/system.slice/app_name.service/libpod-payload-cc316b3065798e96c30dcfc708f7a7dda1b2d78e552364002d2e49a3100ef3ef`

as well, would be nice if that would also get picked up as podman.

Commit:
add Podman container support

Detect Podman/libpod containers via cgroup patterns:
- rootful: machine.slice/libpod-<ID>.scope
- rootless: user.slice/.../libpod-<ID>.scope
- --cgroups=split: system.slice/<svc>/libpod-payload-<ID>
- conmon processes are filtered (like crio-conmon)

Use Podman REST API via Unix socket for container inspection,
following the same http.Client pattern as CRI-O.

Log collection supports both journald (Podman default) and
json-file log drivers via the existing journald reader and
file tail reader respectively.

Authored by LastCoder | e6d8e0a6-31aa-49de-8456-dc6577890436
